### PR TITLE
Indent long `ConditionalExpression` in `MemberExpression`

### PR DESCRIPTION
**Pull request by fisker**
Fixes: #4976

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/method-chain/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -507,21 +507,24 @@ object[valid
...
   .map()
   .filter();
 
-(valid
-  ? helper.responseBody(this.currentUser)
-  : helper.responseBody(defaultUser)
+(
+  valid
+    ? helper.responseBody(this.currentUser)
+    : helper.responseBody(defaultUser)
```
Comment (thorn0) [inline tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: Do we really want to change the formatting in these simpler cases where the condition fits on one line?

Comment (fisker) [reply tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: You mean [this](https://deploy-preview-7564--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAKAzgCwgJxgfS1wKL0OzNJgAIB+aq84pyikt6pBtlg-fAJQAdKADoAtgEMADqgEgANCAjSYAS2jpkoSThwQA7gAVdCLSkkAbA5ICeWpQCMcksAGs4MAMrTXaqADmyDA4AK5wSpgw4pYA6phq8Oi+YHBeZolqAG6Jtsjg6A4g-uhweEYuAVLIAGZWpUoAVugAHgBCLu6eXpLicAAy-nC19REgzS1e-gGWcACKoRDwI5YNIL44pTj5jpKOcJaK6zj+MLFqACYwmMgAHAAMStL6pbEu0vnPcFtZw0oAjot4BUVOYQJJ0ABaKBwOAXOFHHBwQFqJEVSRVSQrNalcRqYJhMboaazBZLYZIOqrMYwPbnK43JAAJiUIUkaks0wAwhBxNUUN8AKxHUKlAAqe3MVLWWXCAEkoPDYF4wCdVABBRVeGC2WbYuAAXwNQA)?

Comment (fisker) [reply tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: Seems I didn't change much, only difference is one line after `(`.

[#7564](https://deploy-preview-7564--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAKAzgCwgJxgAgH58tcCkTs8B9Um6hxp5l1t1gSgB0oA6AWwCGAB1QcA3Dx4YqBYnXKUytWezXqNjbnyGiJUqDLJElefBQUrlmm7bbaBIseJAAaEBGEwAltHTJQQRwcCAB3AAUghH8UQQAbUMEAT393ACMcQTAAazgYAGVhLO8oAHNkGBwAVzh3TBh+OIB1TG94dCKwOHzotu8ANzak5HB0VJAS9Dg8cMzSoWQAM3ip9wArdAAPACFMnLz8wX44ABkSuCWV2pANzfyS0ri4AEUqiHhLuNWQIpwpnBGaUEaTgcTcPxwJRgTW8ABMYJhkAAOAAM7mEISmTUywhGGLg-36F3cAEc3vBZp4YiBBOgALRQOBwWHM8E4OBk7zs2aCeaCT7fKb8bwVarXdAPJ6vd4XJDLL7XGDAmHwxFIABM7kqgm8cQeAGEIPwFigCQBWcFVKYAFWBMXl336NQAklAWbB8mBIV4AIJu-IwJJPAVwAC+oaAA)

[stable](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAKAzgCwgJxgAgH58tcCkTs8B9Um6hxp5l1t1gSgB0oA6AWwCGAB1QcA3Dx4YqBYnXKUytWezXqNjbnyGiJUqDLJElefBQUrlmm7bbaBIseJAAaEBGEwAltHTJQQRwcCAB3AAUghH8UQQA3CG8AEzcQACMcQTAAazgYAGVhLO8oAHNkGBwAVzh3TBh+ABsAdUxveHQisDh86PbvOPaAT2RwdH93EvQ4PHDM0qFkADNBRun3ACt0AA8AIUycvPzBfjgAGRK4ZdX1kC3t-JLSxrgARSqIeGu12pAinGmOFGaUEaTgjVSwhwJRgzWSMEwyAAHAAGdxQiDTZqZYSjKFwQFxK7uACOH3gc08MRAgnQAFooHA4ElmakcHAyd52XNBAtBN9btN+N4KtVfugni93p8rkgVj93DBQXCkgjkAAmRWZbyNJ4AYQg-EWKCg0GJICq0wAKqCYvLbnEagBJKAs2D5MDQrwAQVd+RgQxeArgAF8Q0A)

Comment (thorn0) [reply tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: I mean [this](https://deploy-preview-7564--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAKAbgQwDYEsAmAOlAAQkD8JAFnNgA5wBOAdI3AM53TtwBCE+AJ6oYVXO2ZgArozawAqj0YBKYmSTVaDFm07c+A4aPHN8cAGaYp2GIqaqoy5uYgQQAGhAQ6MXN2SgmLIQAO4ACkEI7MggOCGYgtGeAEaMmGAA1nAwAMp06bhQAObIMIxScJ5UMAC22ADqYvCc6XA5Ubi+6J2CMWDsSSCFSjBhaUU1mMiW2DyeAFbsAB68aZnZOZg1cAAyhXDTOHMgi0s5hUXYcACKUhDwh7OVIPmMSjHJmMm0Hi+MhTB6gRRMgABwABk8dEYEB49TSdBi0I4THQB08AEc7vAxt5oihMOwALRQOBwMz4X5sLG4NhjTATKZIGbHHg1XClcrPdgXK63e4HZlHZ4wL5A-AgpAAJk8ZUwuDwxQAwhAapMYhwAKy-KQ8AAqX3xLOe6AqAEkoGZYDkwP8fABBS05GCCK6PHgAXw9QA). Looks more questionable  when it's [part of expression](https://deploy-preview-7564--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuc0DOMAEBXTBeTARgAZjMBqTACgDcBDAGwEsATAHSk0wH5MALOAwAOcAE4A6UXDRD0cAEIQWATyow+TNOLDZRU2AFU0YgJQcuSfoJESpMuYpVqNWlnABmdbAxhHTHE3F3CAgQABoQCCEYJnRkUDo9CAB3AAVEhDRkEEZkumUsiIAjUTowAGs4GABlITKmKABzZBhRbDgIvhgAWwYAdQ14GTK4asymGJoJ5WywNEKQBuNRGFTSxu66ZE8GYwiAKzQAD3lSiqrqum64ABkGuG3GPZBDo+qGxoY4AEVsCHhHrsOiA6qJltkinQioJwiDRA0YH1WOpkAAOYgRISiCDGPqlITZLHSMQ0B4RACOf3gayiWRQdDQAFooHA4G4WLCpJSmFI1nQNlskDtnsZukwWm1gWgPl9fv8HkKnsCYFCkSwUUgAEwRVp0JjMJoAYQg3U22WkAFZYdhjAAVKF04XAmjtACSUDcsGqYHh0QAgh7qjBlF9AcYAL7hoA).

Comment (thorn0) [reply tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: Also this problem exists [in `return` statements](https://deploy-preview-7564--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAnOMCuqoAIBuAhgDYCWAJrgLw24DmGAyhALYYAWpUdAYqq8zYB1dnHQAKAJS4APjNwxOAZwB0YbOlgBVJWOq0A5DwgQARmJi4AQoVTnUdAwB08uAPy5RxAA5iV6JW9oXSsIcgBPcUVSVXVUTRgdMUkXXFwkTzgfPwCgqBCwyOjVcjgAM0JMYkTdVBSoEAAaEAhvGFJg5FBbfgB3AAVbBCVkEBJewnCR5tNUQjAAayZvea46ZBhUTDhm9hgWYhFSeED5uEZh49J8Y-DRsCVpkC5amH65uhZCZAriXWaAFZKAAeVjmiyYhDYABkuHAfiR-iAgcDGGtiHAAIqYCDwBF-HYgFaoWqjUyEczEJpE1BcGBCCiKZAADgADM1vPxdEI5t5Rpy4LV8PDmgBHHHwd6tEYoQhKAC0UDgcFK5Gp6HFpHQ70In2+SF+SN0LFIGy2hKU6KxEvhBsRhJgFIZ5CZSAATM1NoRSGRuABhVhfUaCgCs1MwugAKhSZYbCfhtgBJKClWCMMC0toAQRTjBg4Qx+N0AF9i0A). Can be solved by adding parens. Or it's probably better to find a generic way to increase the indentation of `?` and `:` if the condition (the part before `?`) doesn't fit on one line.

Comment (fisker) [reply tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: Personal I prefer new one, we are not running `+` with the ternary, we are calculating with `(ternary).foo`

Comment (fisker) [reply tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: Like this?

```js
function a() {
  return (
      valid === getSomethingFromSomeWhere() || this.currentUser === 'Foobert Barberg'
    )
    ? valid === getSomethingFromSomeWhere() ||
       this.currentUser === "Foobert Barberg"
    : valid === getSomethingFromSomeWhere() ||
       this.currentUser === "Foobert Barberg"
}
```

This seems weird....

Comment (thorn0) [reply tests/method-chain/__snapshots__/jsfmt.spec.js.snap]: That's what I meant:
```js
function oneLineCondition() {
  return valid === getSomethingFromSomeWhere()
    ? helper.responseBody(this.currentUser)
    : helper.responseBody(this.defaultUser);
}

function twoLineCondition() {
  return valid === getSomethingFromSomeWhere() ||
    this.currentUser === "Foobert Barberg"
      ? helper.responseBody(this.currentUser)
      : helper.responseBody(this.defaultUser);
}
```

**Comment by fisker** [comment]
Close this , we'll add intention for some cases in #10187

**Pull request closed without merging**